### PR TITLE
[WFCORE-6815][Community] Utility to reload a Host Controller to a different stab…

### PR DESCRIPTION
**Pull request by yersan**
…ility level in the testsuite

Jira issue: https://issues.redhat.com/browse/WFCORE-6815

### CODE REVIEW

**Comment by yersan** [comment]
Hello @kabir , any chance you can take a look at this one as SME for a community feature?
@jamezp Would you agree to be the external third-party contributor and review this one from your side?

**Comment by jamezp** [comment]
@yersan Yes, but as an FYI I'm on PTO for the a week starting Wednesday so if this is time sensitive, I might not be that helpful :)

**Comment by yersan** [comment]
Hi @kabir / @jamezp Any chance to look at this proposal? 

The lack of ability to segregate the tests is blocking work for the conversion branches, since we need to properly avoid the execution of tests that are not meant for the default stability level. If possible, I would like to unlock this. Thanks!

File: server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc
Comment (jamezp) [inline server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc:83]: Should this say the "restart-servers parameter will not be allowed"? I'm asking because the first part of the sentence makes it not sound useful.

If the wording is intended, is the host controller meant to act as the minimum stability level for servers?

Comment (yersan) [reply server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc:83]: @jamezp 

> Should this say the "restart-servers parameter will not be allowed"?

It is currently allowed, but it does not make sense to reload the host to a certain stability level and not restart your servers. If you do that, you could have your host running at one stability level, but your server running at another. I think that could be a problem. Maybe the simplest approach here would be to remove the `restart-servers` attribute from the `reload-enhanced` operation and restart the servers always when this operation is used.

I initially wanted to avoid these differences and assume you should know what you are doing when using the `reload-enhanced`, after all it is for our testsuite, That would bring more flexibility, but maybe that's not a good idea. WDYT?


> is the host controller meant to act as the minimum stability level for servers?

The Host Controller sends its stability level to the managed servers with the managed server boot command, that's all. It dictates what stability level the managed server will run once it gets restarted. So, if you do not restart your servers, you could have the Host Controller running on `preview` and the managed servers still on `community`, I guess that could be problematic somehow, but I'm not sure.

Maybe we should start as restrictive as we can, e.g. remove the restart servers from reload enhanced and always restart them to avoid problems, and if there is a need, open it to allow restarting the server to get more flexibility.

Comment (jamezp) [reply server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc:83]: @yersan IMO then I think the `restart-servers` attribute should be removed from the new operation. I can understand not wanting the difference, However, attributes on operations that don't exist are ignored so it won't really hurt anything and makes it safer I think.

Comment (yersan) [reply server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc]: @jamezp I've removed the restart-servers attribute from the reload-enhanced operation. Let me know how it looks like now

File: server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc
Comment (jamezp) [inline server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc:73]: Very minor, these should be `[ ]` so they show up as checkboxes. Same with the first one too... ...again minor :)

**Review by jamezp — APPROVED** [review]

File: server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc
Comment (yersan) [reply server/WFCORE-6815-reload-domain-mode-to-stability-level.adoc]: Done, thanks for reviewing this @jamezp

**Review by jamezp — APPROVED** [review]

**Comment by yersan** [comment]
@kabir  Let me know if there is anything else to add here, otherwise, approve if you agree, thanks!

**Comment by yersan** [comment]
@kabir Friendly remainder, this one is waiting for your approval to get it merged, please take a look, thanks!

**Review by kabir — APPROVED** [review]

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @yersan, @kabir and @jamezp

**Pull request merged**
